enterprise/views.py

```diff
@@ -226,299 +242,133 @@ def get_default_context(self, enterprise_customer, request):
-            'course_specific': True,
-            'enrollment_deferred': enrollment_deferred is not None,
-            'failure_url': failure_url,
-            'requested_permissions': [
-                _('your enrollment in this course'),
-                _('your learning progress'),
-                _('course completion'),
-            ],
-            'enterprise_customer': customer,
-            'welcome_text': self.welcome_text.format(platform_name=platform_name),
-            'enterprise_welcome_text': self.enterprise_welcome_text.format(
-                enterprise_customer_name=customer.name,
-                platform_name=platform_name,
-                strong_start='<strong>',
-                strong_end='</strong>',
-            ),
-            'policy_link_template': '',
-        }
-        context_data.update(course_specific_context)
-
-        return render(request, 'enterprise/grant_data_sharing_permissions.html', context=context_data)
-
-    @method_decorator(login_required)
-    def get_program_specific_consent(self, request, program_uuid):
+    def get(self, request):
         """
-        Render a form in order to retrieve program-related consent.
+        Render a form to collect user input about data sharing consent.
         """
-        enterprise_uuid = request.GET.get('enterprise_customer_uuid')
+        enterprise_customer_uuid = request.GET.get('enterprise_customer_uuid')
         success_url = request.GET.get('next')
         failure_url = request.GET.get('failure_url')
-        enrollment_deferred = request.GET.get('enrollment_deferred')
-        username = request.user.username
-
-        if not (enterprise_uuid and failure_url and success_url):
+        if not (enterprise_customer_uuid and success_url and failure_url):
             raise Http404
 
-        if not CourseCatalogApiServiceClient.program_exists(program_uuid):
+        course_id = request.GET.get('course_id', '')
+        program_uuid = request.GET.get('program_uuid', '')
+        if not self.course_or_program_exist(course_id, program_uuid):
             raise Http404
 
-        consent_record = get_data_sharing_consent(username, enterprise_uuid, program_uuid=program_uuid)
+        consent_record = get_data_sharing_consent(
+            request.user.username,
+            enterprise_customer_uuid,
+            program_uuid=program_uuid,
+            course_id=course_id
+        )
         if consent_record is None or not consent_record.consent_required():
             raise Http404
 
-        customer = consent_record.enterprise_customer
-
-        context_data = self.get_default_context(customer, request)
-        platform_name = context_data['platform_name']
-
-        # Translators: bold_start and bold_end are HTML tags for specifying
-        # enterprise name in bold text.
-        program_specific_context = {
+        # Translators: bold_start and bold_end are HTML tags for specifying enterprise name in bold text.
```

Comment:
Does this line need to go somewhere else?

Reply:
I guess right above the last update to `context_data`, I put it here because it looked cleaner in the flow. But probably safer to just put it down there, yeah.

Reply:
Yeah, for translation notes, I think they need to be directly above the translation command.